### PR TITLE
Add Example codeblock to the docstrings of important functions in the dataset module

### PR DESCRIPTION
**Pull request by Steven-Yiran**
Added a code example to the docstring for the `find_overlapping_classes` method in the cleanlab.dataset module.

The code block is not currently runnable because of the arbitrary `get_data_labels_from_dataset(yourFavoriteDataset)`. I can define the `get_data_labels_from_dataset` using a simple dataset, but I am not sure where to put this function since other docstring examples will also benefit from having this function.

Before addition:
before
After addition:
after

### CODE REVIEW

File: cleanlab/dataset.py [hunk 1]
```diff
@@ -139,6 +139,27 @@ def find_overlapping_classes(
...
+            data,
+            labels,
+            cv=num_crossval_folds,
+            method="predict_proba",
+        )
+    >>> df = find_overlapping_classes(
+            labels=labels,
+            pred_probs=pred_probs,
+        )
+    >>> df # output pairs of classes that are often mislabeled
```
Comment (jwmueller) [inline cleanlab/dataset.py:161]:
```suggestion
    >>> df = find_overlapping_classes(
            labels=labels,
            pred_probs=pred_probs,
        )  # output pairs of classes that are often mislabeled
```

Comment (jwmueller) [reply cleanlab/dataset.py:161]: don't think we need to include the display commands in examples

Comment (Steven-Yiran) [reply cleanlab/dataset.py:161]: make sense!

**Review by jwmueller — APPROVED** [review]
Thanks for your contribution!  Feel free to add more nice examples like this!

**Comment by jwmueller** [comment]
And remember: once we have a few example codeblocks in the docstrings, it would be nice to add testing capabilities in CI that automatically run these example codeblocks (as additional tests). Eg. using some tool like: https://www.sphinx-doc.org/en/master/usage/extensions/doctest.html

**Pull request merged**